From ntanibata@jp.adit-jv.com Fri Dec 12 03:56:24 2014
Received: from mail.windriver.com (147.11.1.11) by ALA-HCA.corp.ad.wrs.com
 (147.11.189.50) with Microsoft SMTP Server (TLS) id 14.3.174.1; Thu, 11 Dec
 2014 18:56:23 -0800
Received: from mr1.windriver.com (mr1 [192.124.127.25])	by mail.windriver.com
 (8.14.9/8.14.5) with ESMTP id sBC2uLlr026848	(version=TLSv1/SSLv3
 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK);	Thu, 11 Dec 2014 18:56:22
 -0800 (PST)
Received: from esa2.edoxs1.iphmx.com (esa2.edoxs1.iphmx.com [68.232.135.254])
	by mr1.windriver.com (8.14.9/8.14.9) with ESMTP id sBC2uL7v028404
	(version=TLSv1/SSLv3 cipher=RC4-SHA bits=128 verify=FAIL);	Thu, 11 Dec 2014
 18:56:21 -0800
X-IronPort-Anti-Spam-Filtered: true
X-IronPort-Anti-Spam-Result: AtUAAKJYilSFwLVOnGdsb2JhbABZg1hYw1aCGwqFbgKBFRYBAQEBAREBAQEBAQYNCQkULoQMAQEBBAECCRsJEAEBLAsBCwQCAQgRBAEBAQoGFwEGAR4JBBoJCAIEEwgMB4gQDsBZhVIBBZFGAQEBAQEBAQEBAQEBAQEBAQEBAQEBEQaPDwEQAgEeBisHBoInTB2BE4Uig3YohEQEgzqBQIFcIoFzgQwNI4ItihODN4QdXwGBAiSBHAEBAQ
X-IPAS-Result: AtUAAKJYilSFwLVOnGdsb2JhbABZg1hYw1aCGwqFbgKBFRYBAQEBAREBAQEBAQYNCQkULoQMAQEBBAECCRsJEAEBLAsBCwQCAQgRBAEBAQoGFwEGAR4JBBoJCAIEEwgMB4gQDsBZhVIBBZFGAQEBAQEBAQEBAQEBAQEBAQEBAQEBEQaPDwEQAgEeBisHBoInTB2BE4Uig3YohEQEgzqBQIFcIoFzgQwNI4ItihODN4QdXwGBAiSBHAEBAQ
X-IronPort-AV: E=Sophos;i="5.07,561,1413270000"; 
   d="scan'208";a="110097519"
Received: from ma-dnext03.denso.co.jp ([133.192.181.78])  by
 esa2.edoxs1.iphmx.com with ESMTP; 11 Dec 2014 18:56:12 -0800
Received: from guardw2.denso.co.jp (unknown [133.192.24.24])	by
 ma-dnext03.denso.co.jp (Postfix) with ESMTP id 25D295D02BA;	Fri, 12 Dec 2014
 11:56:09 +0900 (JST)
DKIM-Signature: v=1; a=rsa-sha256; c=simple/simple; d=adit-jv.com;
	s=jpadit-jvmail2011; t=1418352969;
	bh=AfmD+uPkOMv42mZQfFad+rgrE601l5DmedvTjy11d4c=;
	h=MIME-Version:Content-Type:Content-Transfer-Encoding:Subject:Date:
	 Message-ID:In-Reply-To:References:From:To:Cc;
	b=V6gBLvmpZtG28BTmzIocIVwltzkvm1YCiUgCQ+FFRQ6rUVlsPjLbJ7OGQP3Z47xQT
	 BveplQwKTCpnNCHqvvqyCNQcuTWbOBjOpgCzkOQH8163Mi6fHeRsrOIWVwvb5WLLbQ
	 Otpd6b/gvSfgv5uBk3cUQHNeG2aby+iZmdRsaQILqkhv1oUlT7FWBP1hHCLGHpqJL4
	 p4MxHtAqKjQlnDJELSK/8ExdbQJGlDsAc0DNk4RzzwUp4Ff3m3CMwRBmkvtma4UVQh
	 NYbSXaPGDnJcYSg7r9qT5k7K1mbBuoKcz0bNYMtTDUsBOLEirXT0kYWqoFfl+Jaw/q
	 eY6yHT+Z0lw2A==
Received: by guardw2.denso.co.jp (Postfix, from userid 0)	id 15D6E6DF3; Fri,
 12 Dec 2014 11:56:09 +0900 (JST)
Received: from smtp1.denso.co.jp [133.192.24.88] 	 by guardw2 with ESMTP id
 MAA08496;	 Fri, 12 Dec 2014 11:56:08 +0900
Received: from msj2.adit-jv.com ([10.71.113.25])	by smtp01.denso.co.jp (MOS
 4.2.4-GA)	with ESMTP id CJC78590;	Fri, 12 Dec 2014 11:56:07 +0900
X-MimeOLE: Produced By Microsoft Exchange V6.5
Content-Class: urn:content-classes:message
Content-Type: text/plain; charset="iso-8859-1"
Content-Transfer-Encoding: quoted-printable
Subject: RE: weston-ivi-shell input method not supported
Date: Fri, 12 Dec 2014 11:56:06 +0900
Message-ID: <EC24A990E5DA3D41A67C0F9C755F987BEA7A60@msj2.adit-jv.com>
In-Reply-To: <0994C93624B9194C867A2AA78BFAA734AD74977C@ALA-MBB.corp.ad.wrs.com>
X-MS-Has-Attach:
X-MS-TNEF-Correlator:
Thread-Topic: weston-ivi-shell input method not supported
Thread-Index: AdAEo9qzd7B8JvN8SiCG1HHmuOgOQgAps/xQAgi50BAAHfTugAAGiR4wACG0O/AAJFPVgAAfTuMAABx/dYABEnjvXgABCAagAFhzyOA=
References: <0994C93624B9194C867A2AA78BFAA734AD7327F7@ALA-MBB.corp.ad.wrs.com> <EC24A990E5DA3D41A67C0F9C755F987BE562C8@msj2.adit-jv.com> <0994C93624B9194C867A2AA78BFAA734AD73F017@ALA-MBB.corp.ad.wrs.com> <EC24A990E5DA3D41A67C0F9C755F987BEA73BB@msj2.adit-jv.com> <0994C93624B9194C867A2AA78BFAA734AD73F824@ALA-MBB.corp.ad.wrs.com> <EC24A990E5DA3D41A67C0F9C755F987BEA7445@msj2.adit-jv.com> <547ECC2F.5000201@windriver.com> <EC24A990E5DA3D41A67C0F9C755F987BEA7500@msj2.adit-jv.com>,<54805D89.5010003@windriver.com> <7A688B48FC95D148BFE83E098348BE505CF002FB@ALA-MBA.corp.ad.wrs.com> <0994C93624B9194C867A2AA78BFAA734AD74977C@ALA-MBB.corp.ad.wrs.com>
From: "Tanibata, Nobuhiko (ADITJ/SWG)" <ntanibata@jp.adit-jv.com>
To: =?iso-8859-1?Q?Bourcier=2C_Fr=E9d=E9ric?=
	<Frederic.Bourcier@windriver.com>, "Behrens, Holger"
	<Holger.Behrens@windriver.com>, "Sarbu, Florin-Ionut (Florin)"
	<Florin.Sarbu@windriver.com>, "Gherzan, Andrei"
	<Andrei.Gherzan@windriver.com>
CC: <genivi-ivi-layer-management@lists.genivi.org>,
	<philippe.robin@technoveo.com>, "Ishikawa, Tetsuri (ADITJ/SWG)"
	<tishikawa@jp.adit-jv.com>
Return-Path: ntanibata@jp.adit-jv.com
X-MS-Exchange-Organization-AuthSource: ALA-HCA.corp.ad.wrs.com
X-MS-Exchange-Organization-AuthAs: Anonymous
MIME-Version: 1.0
Status: RO
Content-Length: 10061
Lines: 284

Hi Frederic-san, Holger-san,

I think I find root cause as result of debug.

For first check, please apply this patch on ivi-shell, branch genivi-1.2.0;=
 I think you are using now.
Please forget genivi-1.3.0 in my another mail. Formally we should use 1.3.0=
 but for demo purpose first fix might be better.

-----------------------------------------------------------------------------------------------
Subject: [PATCH] Add the processing which has notice of click event from compositor

Signed-off-by: Nobuhiko Tanibata <NOBUHIKO_TANIBATA at xddp.denso.co.jp>
---
 ivi-shell/ivi-shell.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/ivi-shell/ivi-shell.c b/ivi-shell/ivi-shell.c
index 468e66a..9238e93 100644
--- a/ivi-shell/ivi-shell.c
+++ b/ivi-shell/ivi-shell.c
@@ -413,6 +413,20 @@ surface_configure_notify(struct wl_listener *listener, void *data)
 
 }
 
+static void
+click_to_activate_binding(struct weston_seat *seat, uint32_t time, uint32_t button,
+                          void *data)
+{
+    if (seat->pointer->grab != &seat->pointer->default_grab) {
+        return;
+    }
+    if (seat->pointer->focus == NULL) {
+        return;
+    }
+
+    weston_surface_activate(seat->pointer->focus->surface, seat);
+}
+
 WL_EXPORT int
 module_init(struct weston_compositor *compositor,
             int *argc, char *argv[])
@@ -479,6 +493,8 @@ module_init(struct weston_compositor *compositor,
 
     keyboard_key_func = ivi_layout->grab_keyboard_key;
 
+    weston_compositor_add_button_binding(compositor, BTN_LEFT, 0, click_to_activate_binding, shell);
+
     free(setting.ivi_module);
     return 0;
 }
1.7.11.7

BR,
Nobuhiko Tanibata
